### PR TITLE
fixed to free issue.

### PR DESCRIPTION
**Pull request by yoya**
- ImageMagick free function.
  - RelinquishMemory (from magick_wand.go : relinquishMemory)
  - RelinquishMemoryCStringArray

- remove free 2 callers
  - cStringArrayToStringSlice
  - sizedCStringArrayToStringSlice

- replace C.free to RelinquishMemory, memory resources by such methods as ImageMagick API

- etc.
   - add RelinquishMemory, RelinquishMemoryCStringArray for memory leak route.

reference) https://github.com/gographics/imagick/pull/37

### CODE REVIEW

**Comment by justinfx** [comment]
Nice! Thanks for the pull request. I will take a look at this, hopefully tonight. If not, tomorrow!

File: imagick/memory.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#include <wand/MagickWand.h>
+*/
+import "C"
+
+import (
+	"unsafe"
+)
+
+// Relinquishes memory resources returned by such methods as MagickIdentifyImage(), MagickGetException(), etc
+func RelinquishMemory(ptr unsafe.Pointer) {
```
Comment (justinfx) [inline imagick/memory.go]: Should this function, alone with `RelinquishMemoryCStringArray` be package private? Is there any reason we can think of for users of the package to need to free unsafe pointers? My guess (and maybe I am wrong) is that we are trying to hide the details of the memory management.

**Comment by yoya** [comment]
I didn't notice and agree > hide the details of the memory management.
Function uncapitalise is needed.
I make new PR follow this.

**Comment by justinfx** [comment]
Can you just commit changes to your same merge branch? No need for new pull requests. I can do a clean squash merge. Just commit and push the changes. Thanks!

**Comment by yoya** [comment]
Ok. I'll make new commit in same branch, and i think that
not only RelinquishMemoryCStringArray but also RelinquishMemory replacing is needed.

**Comment by yoya** [comment]
commit is here.
https://github.com/yoya/imagick/commits/fixtofreePR
=> https://github.com/yoya/imagick/commit/8b085901627155840ff91b356d68d98a4877cd0b

**Comment by yoya** [comment]
Thank you very much!

**Comment by justinfx** [comment]
awesome work!

**Pull request merged**